### PR TITLE
add support for creating a route53 alias for custom domain

### PR DESCRIPTION
**Pull request by grimm26**
also be explicit about using the provider in this module which avoids trouble in the future.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.idea
+vendor
+.terraform
+.vscode
+*.tfstate
+*.tfstate.backup
```
Comment (justincampbell) [inline .gitignore]: Thanks for adding a gitignore! Could you please remove non-Terraform files from it? You can ignore these globally if you prefer with:

```
echo ".idea" >> ~/.gitignore
echo ".vscode" >> ~/.gitignore
git config --global core.excludesfile ~/.gitignore
```

File: vars.tf [hunk 2]
```diff
@@ -15,3 +15,12 @@ variable "region" {
...
   default     = "us-east-1"
 }
+
+variable "enable_route53_custom_domain" {
+  description = "Boolean for whether to create an alias in route53 for bucket."
+  default     = false
+}
+
+variable "route53_zone" {
+  description = "AWS Route53 zone name for custom domain name for bucket."
```
Comment (justincampbell) [inline vars.tf]: This is a required variable, even though it is never used if `enable_route53_custom_domain` remains defaulted to false. Can we make this default to an empty string?

**Review by justincampbell — CHANGES_REQUESTED** [review]
Thanks so much for working on this! Other than the required variable mentioned above, I got a clean plan when trying with my current use cases, so I'm happy to merge this after addressing that.

@grimm26 Could you also update the README with documentation about the Route 53 variables/functionality?

File: vars.tf (hunk 2 above)
Comment (grimm26) [reply vars.tf]: I also noticed that you have an unused variable, domain, that becomes confusing with my feature in place.  Can I just remove it?

**Pull request closed without merging**